src/client.js

```diff
@@ -476,6 +476,119 @@ class ConnectionParameters {
 
 }
 
+// ========================================================================================
+// File Uploader
+// ========================================================================================
+
+const fileUploader = (client) => {
+  /**
+   * Will increament the counter
+   * @returns {Promise<number|*>}
+   */
+  const increaseValue = async () => {
+    const soapCall = client._prepareSoapCall('xtk:counter', 'IncreaseValue')
+    soapCall.writeString('name', 'xtkResource')
+    await client._makeSoapCall(soapCall)
+    return soapCall.getNextLong();
+  }
+
+  /**
+   * Will write fileRes
+   * @param counter
+   * @param data
+   * @returns {Promise<*>}
+   */
+  const write = async (counter, data) => {
+    const soapCall = client._prepareSoapCall('xtk:persist', 'Write');
+    const fileRes = soapCall.createElement("fileRes");
+    fileRes.setAttribute("internalName", "RES" + counter);
+    fileRes.setAttribute("md5", data[0].md5);
+    fileRes.setAttribute("label", data[0].fileName);
+    fileRes.setAttribute("fileName", data[0].fileName);
+    fileRes.setAttribute("originalName", data[0].fileName);
+    fileRes.setAttribute("useMd5AsFilename", "1");
+    fileRes.setAttribute("storageType", "5");
+    fileRes.setAttribute("xtkschema", "xtk:fileRes");
+    soapCall.writeDocument("doc", DomUtil.parse(DomUtil.toXMLString(fileRes)));
+    await client._makeSoapCall(soapCall)
+    return fileRes;
+  }
+
+  /**
+   * Will publish the file
+   * @param fileRes
+   * @returns {Promise<void>}
+   */
+  const publishIfNeeded = async (fileRes) => {
+    const soapCall = client._prepareSoapCall('xtk:fileRes', 'PublishIfNeeded');
+    soapCall.writeDocument("entity", DomUtil.parse(DomUtil.toXMLString(fileRes)));
+    await client._makeSoapCall(soapCall)
+  }
+
+  /**
+   * Will return the public URL of the uploaded file.
+   * @param fileRes
+   * @returns {Promise<string|*>}
+   */
+  const getPublicUrl = async (fileRes) => {
+    const soapCall = client._prepareSoapCall('xtk:fileRes', 'GetURL');
+    soapCall.writeDocument("entity", DomUtil.parse(DomUtil.toXMLString(fileRes)));
+    await client._makeSoapCall(soapCall)
+    soapCall.getNextElement();
+    return soapCall.getNextString();
+  }
+  return {
+    upload: async (file) => {
+      return new Promise(async (resolve, reject) => {
+        if (!Util.isBrowser()) {
+          reject('File uploading is only supported in browser based calls.');
+        }
+        try {
+          var data = new FormData()
+          data.append('file_noMd5', file)
+          //TODO: Needs to be refactored after cookie issue get resolved.
+          const response = await fetch(`${client._connectionParameters._endpoint}/nl/jsp/uploadFile.jsp`, {
```

Comment:
Did you tried using **request** defined in  https://github.com/adobe/acc-js-sdk/blob/master/src/transport.js